File: src/frontends/cli/query.py
import pickledb, os
from utils import debug, error

class Query:

    def __init__(self):
        self.db = pickledb.load(os.environ["HOME"] + "/.dent/hooks.db", True)
        print(self.db.get("Pos"))
    
    def get_hook_value(self, name):
        """ Returns value from the database given a name """
        debug("DB: GET " + str(name))
        return self.db.get(name)

    def set_hook_value(self, name, value):
        """ Update's a hook's value in the database given its name and new value """
        self.db.set(name, str(value))
        debug("DB: SET " + name + ", " + str(value))
    
    def delete_hook(self, name):
        """ Deletes hook data given a name """
        self.db.rem(str(name))
        debug("DB: DEL " + name)

File: src/backends/base-java/src/main/java/dent/backends/base/Query.java
package dent.backends.base;

import dent.backends.base.DB;
import dent.backends.base.Utils;

import java.io.FileNotFoundException;

public class Query {

    protected DB db;
    private final Utils utils;

    public Query() {
        this.db = new DB(System.getProperty("user.home") + "/.dent/db.db");
        db.load();
        this.utils = new Utils();
    }

    public String getHookValue(String name) {
        this.utils.debug("DB: GET " + name);
        try {
            return this.db.get(name);
        } catch (FileNotFoundException e) {
            this.utils.error(e.getMessage());
        }
        return null;
    }

    public void setHookValue(String name, String value) {
        this.utils.debug("DB: SET " + name + ", " + value);
        this.db.set(name, value);
    }
}
